src/main/java/net/neoforged/neoforge/common/util/FriendlyByteBufUtil.java

```diff
@@ -31,7 +31,10 @@ public static byte[] writeCustomData(Consumer<RegistryFriendlyByteBuf> dataWrite
         final RegistryFriendlyByteBuf buf = new RegistryFriendlyByteBuf(Unpooled.buffer(), registryAccess, ConnectionType.NEOFORGE);
         try {
             dataWriter.accept(buf);
-            return buf.array();
```

Comment:
Instead of replacing this with your own copy: I am like 90% sure that there is a method that gets you an exact sized array. Could you check?

Reply:
I don't think there is, at least i'm not aware of any.
The only method that is similar is FriendlyByteBuf#readByteArray(). However, this reads the length of the array as VarInt and is therefore not possible to use here.

Reply:
@marchermans I cannot find any method on RegistryFriendlyByteBuf that matches an exact sized array safely. Did you want to take a look too or should we move forward with this PR?

Reply:
This implementation looks fine to me